// @aws-sdk/client-kinesis/dist-cjs/schemas/schemas_0.js
const _AAT = "ApproximateArrivalTimestamp";
const _ADE = "AccessDeniedException";
const _APSI = "AdjacentParentShardId";
const _ASTM = "AdjacentShardToMerge";
const _ATTS = "AddTagsToStream";
const _ATTSI = "AddTagsToStreamInput";
const _C = "Consumer";
const _CARN = "ConsumerARN";
const _CC = "ConsumerCount";
const _CCT = "ConsumerCreationTimestamp";
const _CD = "ConsumerDescription";
const _CL = "ConsumerList";
const _CMB = "CurrentMiBps";
const _CN = "ConsumerName";
const _CS = "ChildShard";
const _CSC = "CurrentShardCount";
const _CSI = "CreateStreamInput";
const _CSL = "ChildShardList";
const _CSLM = "CurrentShardLevelMetrics";
const _CSN = "ContinuationSequenceNumber";
const _CSh = "ChildShards";
const _CSo = "ConsumerStatus";
const _CSr = "CreateStream";
const _Co = "Consumers";
const _D = "Data";
const _DAS = "DescribeAccountSettings";
const _DASI = "DescribeAccountSettingsInput";
const _DASO = "DescribeAccountSettingsOutput";
const _DEM = "DisableEnhancedMonitoring";
const _DEMI = "DisableEnhancedMonitoringInput";
const _DL = "DescribeLimits";
const _DLI = "DescribeLimitsInput";
const _DLO = "DescribeLimitsOutput";
const _DRP = "DeleteResourcePolicy";
const _DRPI = "DeleteResourcePolicyInput";
const _DS = "DeleteStream";
const _DSC = "DeregisterStreamConsumer";
const _DSCI = "DeregisterStreamConsumerInput";
const _DSCIe = "DescribeStreamConsumerInput";
const _DSCO = "DescribeStreamConsumerOutput";
const _DSCe = "DescribeStreamConsumer";
const _DSI = "DeleteStreamInput";
const _DSIe = "DescribeStreamInput";
const _DSLM = "DesiredShardLevelMetrics";
const _DSO = "DescribeStreamOutput";
const _DSRP = "DecreaseStreamRetentionPeriod";
const _DSRPI = "DecreaseStreamRetentionPeriodInput";
const _DSS = "DescribeStreamSummary";
const _DSSI = "DescribeStreamSummaryInput";
const _DSSO = "DescribeStreamSummaryOutput";
const _DSe = "DescribeStream";
const _EA = "EndedAt";
const _EAEA = "EarliestAllowedEndAt";
const _EC = "ErrorCode";
const _ECD = "EnforceConsumerDeletion";
const _EEM = "EnableEnhancedMonitoring";
const _EEMI = "EnableEnhancedMonitoringInput";
const _EHK = "EndingHashKey";
const _EHKx = "ExplicitHashKey";
const _EIE = "ExpiredIteratorException";
const _EM = "EnhancedMetrics";
const _EML = "EnhancedMonitoringList";
const _EMO = "EnhancedMonitoringOutput";
const _EMn = "EnhancedMonitoring";
const _EMr = "ErrorMessage";
const _ENTE = "ExpiredNextTokenException";
const _ES = "EventStream";
const _ESN = "EndingSequenceNumber";
const _ESSI = "ExclusiveStartShardId";
const _ESSN = "ExclusiveStartStreamName";
const _ESTK = "ExclusiveStartTagKey";
const _ET = "EncryptionType";
const _FRC = "FailedRecordCount";
const _GR = "GetRecords";
const _GRI = "GetRecordsInput";
const _GRO = "GetRecordsOutput";
const _GRP = "GetResourcePolicy";
const _GRPI = "GetResourcePolicyInput";
const _GRPO = "GetResourcePolicyOutput";
const _GSI = "GetShardIterator";
const _GSII = "GetShardIteratorInput";
const _GSIO = "GetShardIteratorOutput";
const _HKR = "HashKeyRange";
const _HMS = "HasMoreStreams";
const _HMSa = "HasMoreShards";
const _HMT = "HasMoreTags";
const _IAE = "InvalidArgumentException";
const _IFE = "InternalFailureException";
const _ISRP = "IncreaseStreamRetentionPeriod";
const _ISRPI = "IncreaseStreamRetentionPeriodInput";
const _K = "Key";
const _KI = "KeyId";
const _KMSADE = "KMSAccessDeniedException";
const _KMSDE = "KMSDisabledException";
const _KMSISE = "KMSInvalidStateException";
const _KMSNFE = "KMSNotFoundException";
const _KMSOIR = "KMSOptInRequired";
const _KMSTE = "KMSThrottlingException";
const _L = "Limit";
const _LEE = "LimitExceededException";
const _LS = "ListShards";
const _LSC = "ListStreamConsumers";
const _LSCI = "ListStreamConsumersInput";
const _LSCO = "ListStreamConsumersOutput";
const _LSI = "ListShardsInput";
const _LSIi = "ListStreamsInput";
const _LSO = "ListShardsOutput";
const _LSOi = "ListStreamsOutput";
const _LSi = "ListStreams";
const _LTFR = "ListTagsForResource";
const _LTFRI = "ListTagsForResourceInput";
const _LTFRO = "ListTagsForResourceOutput";
const _LTFS = "ListTagsForStream";
const _LTFSI = "ListTagsForStreamInput";
const _LTFSO = "ListTagsForStreamOutput";
const _MBL = "MillisBehindLatest";
const _MR = "MaxResults";
const _MRSIKB = "MaxRecordSizeInKiB";
const _MS = "MergeShards";
const _MSI = "MergeShardsInput";
const _MTBC = "MinimumThroughputBillingCommitment";
const _MTBCI = "MinimumThroughputBillingCommitmentInput";
const _MTBCO = "MinimumThroughputBillingCommitmentOutput";
const _NSHK = "NewStartingHashKey";
const _NSI = "NextShardIterator";
const _NT = "NextToken";
const _ODSC = "OnDemandStreamCount";
const _ODSCL = "OnDemandStreamCountLimit";
const _OSC = "OpenShardCount";
const _P = "Policy";
const _PK = "PartitionKey";
const _PR = "PutRecord";
const _PRI = "PutRecordInput";
const _PRIu = "PutRecordsInput";
const _PRO = "PutRecordOutput";
const _PROu = "PutRecordsOutput";
const _PRP = "PutResourcePolicy";
const _PRPI = "PutResourcePolicyInput";
const _PRRE = "PutRecordsRequestEntry";
const _PRREL = "PutRecordsRequestEntryList";
const _PRRELu = "PutRecordsResultEntryList";
const _PRREu = "PutRecordsResultEntry";
const _PRu = "PutRecords";
const _PS = "ParentShards";
const _PSI = "ParentShardId";
const _PTEE = "ProvisionedThroughputExceededException";
const _R = "Records";
const _RARN = "ResourceARN";
const _RIUE = "ResourceInUseException";
const _RL = "RecordList";
const _RNFE = "ResourceNotFoundException";
const _RPH = "RetentionPeriodHours";
const _RSC = "RegisterStreamConsumer";
const _RSCI = "RegisterStreamConsumerInput";
const _RSCO = "RegisterStreamConsumerOutput";
const _RTFS = "RemoveTagsFromStream";
const _RTFSI = "RemoveTagsFromStreamInput";
const _Re = "Record";
const _S = "Shards";
const _SA = "StartedAt";
const _SARN = "StreamARN";
const _SC = "ShardCount";
const _SCT = "StreamCreationTimestamp";
const _SD = "StreamDescription";
const _SDS = "StreamDescriptionSummary";
const _SF = "ShardFilter";
const _SHK = "StartingHashKey";
const _SI = "StreamId";
const _SIT = "ShardIteratorType";
const _SIh = "ShardId";
const _SIha = "ShardIterator";
const _SL = "ShardLimit";
const _SLM = "ShardLevelMetrics";
const _SLh = "ShardList";
const _SM = "StreamMode";
const _SMD = "StreamModeDetails";
const _SN = "StreamName";
const _SNFO = "SequenceNumberForOrdering";
const _SNR = "SequenceNumberRange";
const _SNe = "SequenceNumber";
const _SNt = "StreamNames";
const _SP = "StartingPosition";
const _SS = "StreamSummaries";
const _SSE = "StartStreamEncryption";
const _SSEI = "StartStreamEncryptionInput";
const _SSEIt = "StopStreamEncryptionInput";
const _SSEt = "StopStreamEncryption";
const _SSI = "SplitShardInput";
const _SSL = "StreamSummaryList";
const _SSN = "StartingSequenceNumber";
const _SSp = "SplitShard";
const _SSt = "StreamStatus";
const _SStr = "StreamSummary";
const _ST = "ScalingType";
const _STM = "ShardToMerge";
const _STS = "ShardToSplit";
const _STSE = "SubscribeToShardEvent";
const _STSES = "SubscribeToShardEventStream";
const _STSI = "SubscribeToShardInput";
const _STSO = "SubscribeToShardOutput";
const _STSu = "SubscribeToShard";
const _Sh = "Shard";
const _St = "Status";
const _T = "Tags";
const _TK = "TagKeys";
const _TL = "TagList";
const _TMB = "TargetMiBps";
const _TR = "TagResource";
const _TRI = "TagResourceInput";
const _TSC = "TargetShardCount";
const _Ta = "Tag";
const _Ti = "Timestamp";
const _Ty = "Type";
const _UAS = "UpdateAccountSettings";
const _UASI = "UpdateAccountSettingsInput";
const _UASO = "UpdateAccountSettingsOutput";
const _UMRS = "UpdateMaxRecordSize";
const _UMRSI = "UpdateMaxRecordSizeInput";
const _UR = "UntagResource";
const _URI = "UntagResourceInput";
const _USC = "UpdateShardCount";
const _USCI = "UpdateShardCountInput";
const _USCO = "UpdateShardCountOutput";
const _USM = "UpdateStreamMode";
const _USMI = "UpdateStreamModeInput";
const _USWT = "UpdateStreamWarmThroughput";
const _USWTI = "UpdateStreamWarmThroughputInput";
const _USWTO = "UpdateStreamWarmThroughputOutput";
const _V = "Value";
const _VE = "ValidationException";
const _WT = "WarmThroughput";
const _WTMB = "WarmThroughputMiBps";
const _WTO = "WarmThroughputObject";
const _c = "client";
const _e = "error";
const _m = "message";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.kinesis";
const _se = "server";
const _st = "streaming";
const n0 = "com.amazonaws.kinesis";
const { TypeRegistry } = require("@smithy/core/schema");
const { AccessDeniedException, ExpiredIteratorException, ExpiredNextTokenException, InternalFailureException, InvalidArgumentException, KMSAccessDeniedException, KMSDisabledException, KMSInvalidStateException, KMSNotFoundException, KMSOptInRequired, KMSThrottlingException, LimitExceededException, ProvisionedThroughputExceededException, ResourceInUseException, ResourceNotFoundException, ValidationException } = require("../models/errors");
const { KinesisServiceException } = require("../models/KinesisServiceException");
const _s_registry = TypeRegistry.for(_s);
const KinesisServiceException$ = [-3, _s, "KinesisServiceException", 0, [], []];
exports.KinesisServiceException$ = KinesisServiceException$;
_s_registry.registerError(KinesisServiceException$, KinesisServiceException);
const n0_registry = TypeRegistry.for(n0);
const AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.AccessDeniedException$ = AccessDeniedException$;
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
const ExpiredIteratorException$ = [-3, n0, _EIE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ExpiredIteratorException$ = ExpiredIteratorException$;
n0_registry.registerError(ExpiredIteratorException$, ExpiredIteratorException);
const ExpiredNextTokenException$ = [-3, n0, _ENTE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ExpiredNextTokenException$ = ExpiredNextTokenException$;
n0_registry.registerError(ExpiredNextTokenException$, ExpiredNextTokenException);
const InternalFailureException$ = [-3, n0, _IFE,
    { [_e]: _se },
    [_m],
    [0]
];
exports.InternalFailureException$ = InternalFailureException$;
n0_registry.registerError(InternalFailureException$, InternalFailureException);
const InvalidArgumentException$ = [-3, n0, _IAE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.InvalidArgumentException$ = InvalidArgumentException$;
n0_registry.registerError(InvalidArgumentException$, InvalidArgumentException);
const KMSAccessDeniedException$ = [-3, n0, _KMSADE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.KMSAccessDeniedException$ = KMSAccessDeniedException$;
n0_registry.registerError(KMSAccessDeniedException$, KMSAccessDeniedException);
const KMSDisabledException$ = [-3, n0, _KMSDE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.KMSDisabledException$ = KMSDisabledException$;
n0_registry.registerError(KMSDisabledException$, KMSDisabledException);
const KMSInvalidStateException$ = [-3, n0, _KMSISE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.KMSInvalidStateException$ = KMSInvalidStateException$;
n0_registry.registerError(KMSInvalidStateException$, KMSInvalidStateException);
const KMSNotFoundException$ = [-3, n0, _KMSNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.KMSNotFoundException$ = KMSNotFoundException$;
n0_registry.registerError(KMSNotFoundException$, KMSNotFoundException);
const KMSOptInRequired$ = [-3, n0, _KMSOIR,
    { [_e]: _c },
    [_m],
    [0]
];
exports.KMSOptInRequired$ = KMSOptInRequired$;
n0_registry.registerError(KMSOptInRequired$, KMSOptInRequired);
const KMSThrottlingException$ = [-3, n0, _KMSTE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.KMSThrottlingException$ = KMSThrottlingException$;
n0_registry.registerError(KMSThrottlingException$, KMSThrottlingException);
const LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.LimitExceededException$ = LimitExceededException$;
n0_registry.registerError(LimitExceededException$, LimitExceededException);
const ProvisionedThroughputExceededException$ = [-3, n0, _PTEE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ProvisionedThroughputExceededException$ = ProvisionedThroughputExceededException$;
n0_registry.registerError(ProvisionedThroughputExceededException$, ProvisionedThroughputExceededException);
const ResourceInUseException$ = [-3, n0, _RIUE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ResourceInUseException$ = ResourceInUseException$;
n0_registry.registerError(ResourceInUseException$, ResourceInUseException);
const ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
const ValidationException$ = [-3, n0, _VE,
    { [_e]: _c },
    [_m],
    [0]
];
exports.ValidationException$ = ValidationException$;
n0_registry.registerError(ValidationException$, ValidationException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
const AddTagsToStreamInput$ = [3, n0, _ATTSI,
    0,
    [_T, _SN, _SARN, _SI],
    [128 | 0, 0, 0, 0], 1
];
exports.AddTagsToStreamInput$ = AddTagsToStreamInput$;
const ChildShard$ = [3, n0, _CS,
    0,
    [_SIh, _PS, _HKR],
    [0, 64 | 0, () => HashKeyRange$], 3
];
exports.ChildShard$ = ChildShard$;
const Consumer$ = [3, n0, _C,
    0,
    [_CN, _CARN, _CSo, _CCT],
    [0, 0, 0, 4], 4
];
exports.Consumer$ = Consumer$;
const ConsumerDescription$ = [3, n0, _CD,
    0,
    [_CN, _CARN, _CSo, _CCT, _SARN],
    [0, 0, 0, 4, 0], 5
];
exports.ConsumerDescription$ = ConsumerDescription$;
const CreateStreamInput$ = [3, n0, _CSI,
    0,
    [_SN, _SC, _SMD, _T, _WTMB, _MRSIKB],
    [0, 1, () => StreamModeDetails$, 128 | 0, 1, 1], 1
];
exports.CreateStreamInput$ = CreateStreamInput$;
const DecreaseStreamRetentionPeriodInput$ = [3, n0, _DSRPI,
    0,
    [_RPH, _SN, _SARN, _SI],
    [1, 0, 0, 0], 1
];
exports.DecreaseStreamRetentionPeriodInput$ = DecreaseStreamRetentionPeriodInput$;
const DeleteResourcePolicyInput$ = [3, n0, _DRPI,
    0,
    [_RARN, _SI],
    [0, 0], 1
];
exports.DeleteResourcePolicyInput$ = DeleteResourcePolicyInput$;
const DeleteStreamInput$ = [3, n0, _DSI,
    0,
    [_SN, _ECD, _SARN, _SI],
    [0, 2, 0, 0]
];
exports.DeleteStreamInput$ = DeleteStreamInput$;
const DeregisterStreamConsumerInput$ = [3, n0, _DSCI,
    0,
    [_SARN, _CN, _CARN, _SI],
    [0, 0, 0, 0]
];
exports.DeregisterStreamConsumerInput$ = DeregisterStreamConsumerInput$;
const DescribeAccountSettingsInput$ = [3, n0, _DASI,
    0,
    [],
    []
];
exports.DescribeAccountSettingsInput$ = DescribeAccountSettingsInput$;
const DescribeAccountSettingsOutput$ = [3, n0, _DASO,
    0,
    [_MTBC],
    [() => MinimumThroughputBillingCommitmentOutput$]
];
exports.DescribeAccountSettingsOutput$ = DescribeAccountSettingsOutput$;
const DescribeLimitsInput$ = [3, n0, _DLI,
    0,
    [],
    []
];
exports.DescribeLimitsInput$ = DescribeLimitsInput$;
const DescribeLimitsOutput$ = [3, n0, _DLO,
    0,
    [_SL, _OSC, _ODSC, _ODSCL],
    [1, 1, 1, 1], 4
];
exports.DescribeLimitsOutput$ = DescribeLimitsOutput$;
const DescribeStreamConsumerInput$ = [3, n0, _DSCIe,
    0,
    [_SARN, _CN, _CARN, _SI],
    [0, 0, 0, 0]
];
exports.DescribeStreamConsumerInput$ = DescribeStreamConsumerInput$;
const DescribeStreamConsumerOutput$ = [3, n0, _DSCO,
    0,
    [_CD],
    [() => ConsumerDescription$], 1
];
exports.DescribeStreamConsumerOutput$ = DescribeStreamConsumerOutput$;
const DescribeStreamInput$ = [3, n0, _DSIe,
    0,
    [_SN, _L, _ESSI, _SARN, _SI],
    [0, 1, 0, 0, 0]
];
exports.DescribeStreamInput$ = DescribeStreamInput$;
const DescribeStreamOutput$ = [3, n0, _DSO,
    0,
    [_SD],
    [() => StreamDescription$], 1
];
exports.DescribeStreamOutput$ = DescribeStreamOutput$;
const DescribeStreamSummaryInput$ = [3, n0, _DSSI,
    0,
    [_SN, _SARN, _SI],
    [0, 0, 0]
];
exports.DescribeStreamSummaryInput$ = DescribeStreamSummaryInput$;
const DescribeStreamSummaryOutput$ = [3, n0, _DSSO,
    0,
    [_SDS],
    [() => StreamDescriptionSummary$], 1
];
exports.DescribeStreamSummaryOutput$ = DescribeStreamSummaryOutput$;
const DisableEnhancedMonitoringInput$ = [3, n0, _DEMI,
    0,
    [_SLM, _SN, _SARN, _SI],
    [64 | 0, 0, 0, 0], 1
];
exports.DisableEnhancedMonitoringInput$ = DisableEnhancedMonitoringInput$;
const EnableEnhancedMonitoringInput$ = [3, n0, _EEMI,
    0,
    [_SLM, _SN, _SARN, _SI],
    [64 | 0, 0, 0, 0], 1
];
exports.EnableEnhancedMonitoringInput$ = EnableEnhancedMonitoringInput$;
const EnhancedMetrics$ = [3, n0, _EM,
    0,
    [_SLM],
    [64 | 0]
];
exports.EnhancedMetrics$ = EnhancedMetrics$;
const EnhancedMonitoringOutput$ = [3, n0, _EMO,
    0,
    [_SN, _CSLM, _DSLM, _SARN],
    [0, 64 | 0, 64 | 0, 0]
];
exports.EnhancedMonitoringOutput$ = EnhancedMonitoringOutput$;
const GetRecordsInput$ = [3, n0, _GRI,
    0,
    [_SIha, _L, _SARN, _SI],
    [0, 1, 0, 0], 1
];
exports.GetRecordsInput$ = GetRecordsInput$;
const GetRecordsOutput$ = [3, n0, _GRO,
    0,
    [_R, _NSI, _MBL, _CSh],
    [() => RecordList, 0, 1, () => ChildShardList], 1
];
exports.GetRecordsOutput$ = GetRecordsOutput$;
const GetResourcePolicyInput$ = [3, n0, _GRPI,
    0,
    [_RARN, _SI],
    [0, 0], 1
];
exports.GetResourcePolicyInput$ = GetResourcePolicyInput$;
const GetResourcePolicyOutput$ = [3, n0, _GRPO,
    0,
    [_P],
    [0], 1
];
exports.GetResourcePolicyOutput$ = GetResourcePolicyOutput$;
const GetShardIteratorInput$ = [3, n0, _GSII,
    0,
    [_SIh, _SIT, _SN, _SSN, _Ti, _SARN, _SI],
    [0, 0, 0, 0, 4, 0, 0], 2
];
exports.GetShardIteratorInput$ = GetShardIteratorInput$;
const GetShardIteratorOutput$ = [3, n0, _GSIO,
    0,
    [_SIha],
    [0]
];
exports.GetShardIteratorOutput$ = GetShardIteratorOutput$;
const HashKeyRange$ = [3, n0, _HKR,
    0,
    [_SHK, _EHK],
    [0, 0], 2
];
exports.HashKeyRange$ = HashKeyRange$;
const IncreaseStreamRetentionPeriodInput$ = [3, n0, _ISRPI,
    0,
    [_RPH, _SN, _SARN, _SI],
    [1, 0, 0, 0], 1
];
exports.IncreaseStreamRetentionPeriodInput$ = IncreaseStreamRetentionPeriodInput$;
const ListShardsInput$ = [3, n0, _LSI,
    0,
    [_SN, _NT, _ESSI, _MR, _SCT, _SF, _SARN, _SI],
    [0, 0, 0, 1, 4, () => ShardFilter$, 0, 0]
];
exports.ListShardsInput$ = ListShardsInput$;
const ListShardsOutput$ = [3, n0, _LSO,
    0,
    [_S, _NT],
    [() => ShardList, 0]
];
exports.ListShardsOutput$ = ListShardsOutput$;
const ListStreamConsumersInput$ = [3, n0, _LSCI,
    0,
    [_SARN, _NT, _MR, _SCT, _SI],
    [0, 0, 1, 4, 0], 1
];
exports.ListStreamConsumersInput$ = ListStreamConsumersInput$;
const ListStreamConsumersOutput$ = [3, n0, _LSCO,
    0,
    [_Co, _NT],
    [() => ConsumerList, 0]
];
exports.ListStreamConsumersOutput$ = ListStreamConsumersOutput$;
const ListStreamsInput$ = [3, n0, _LSIi,
    0,
    [_L, _ESSN, _NT],
    [1, 0, 0]
];
exports.ListStreamsInput$ = ListStreamsInput$;
const ListStreamsOutput$ = [3, n0, _LSOi,
    0,
    [_SNt, _HMS, _NT, _SS],
    [64 | 0, 2, 0, () => StreamSummaryList], 2
];
exports.ListStreamsOutput$ = ListStreamsOutput$;
const ListTagsForResourceInput$ = [3, n0, _LTFRI,
    0,
    [_RARN, _SI],
    [0, 0], 1
];
exports.ListTagsForResourceInput$ = ListTagsForResourceInput$;
const ListTagsForResourceOutput$ = [3, n0, _LTFRO,
    0,
    [_T],
    [() => TagList]
];
exports.ListTagsForResourceOutput$ = ListTagsForResourceOutput$;
const ListTagsForStreamInput$ = [3, n0, _LTFSI,
    0,
    [_SN, _ESTK, _L, _SARN, _SI],
    [0, 0, 1, 0, 0]
];
exports.ListTagsForStreamInput$ = ListTagsForStreamInput$;
const ListTagsForStreamOutput$ = [3, n0, _LTFSO,
    0,
    [_T, _HMT],
    [() => TagList, 2], 2
];
exports.ListTagsForStreamOutput$ = ListTagsForStreamOutput$;
const MergeShardsInput$ = [3, n0, _MSI,
    0,
    [_STM, _ASTM, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.MergeShardsInput$ = MergeShardsInput$;
const MinimumThroughputBillingCommitmentInput$ = [3, n0, _MTBCI,
    0,
    [_St],
    [0], 1
];
exports.MinimumThroughputBillingCommitmentInput$ = MinimumThroughputBillingCommitmentInput$;
const MinimumThroughputBillingCommitmentOutput$ = [3, n0, _MTBCO,
    0,
    [_St, _SA, _EA, _EAEA],
    [0, 4, 4, 4], 1
];
exports.MinimumThroughputBillingCommitmentOutput$ = MinimumThroughputBillingCommitmentOutput$;
const PutRecordInput$ = [3, n0, _PRI,
    0,
    [_D, _PK, _SN, _EHKx, _SNFO, _SARN, _SI],
    [21, 0, 0, 0, 0, 0, 0], 2
];
exports.PutRecordInput$ = PutRecordInput$;
const PutRecordOutput$ = [3, n0, _PRO,
    0,
    [_SIh, _SNe, _ET],
    [0, 0, 0], 2
];
exports.PutRecordOutput$ = PutRecordOutput$;
const PutRecordsInput$ = [3, n0, _PRIu,
    0,
    [_R, _SN, _SARN, _SI],
    [() => PutRecordsRequestEntryList, 0, 0, 0], 1
];
exports.PutRecordsInput$ = PutRecordsInput$;
const PutRecordsOutput$ = [3, n0, _PROu,
    0,
    [_R, _FRC, _ET],
    [() => PutRecordsResultEntryList, 1, 0], 1
];
exports.PutRecordsOutput$ = PutRecordsOutput$;
const PutRecordsRequestEntry$ = [3, n0, _PRRE,
    0,
    [_D, _PK, _EHKx],
    [21, 0, 0], 2
];
exports.PutRecordsRequestEntry$ = PutRecordsRequestEntry$;
const PutRecordsResultEntry$ = [3, n0, _PRREu,
    0,
    [_SNe, _SIh, _EC, _EMr],
    [0, 0, 0, 0]
];
exports.PutRecordsResultEntry$ = PutRecordsResultEntry$;
const PutResourcePolicyInput$ = [3, n0, _PRPI,
    0,
    [_RARN, _P, _SI],
    [0, 0, 0], 2
];
exports.PutResourcePolicyInput$ = PutResourcePolicyInput$;
const _Record$ = [3, n0, _Re,
    0,
    [_SNe, _D, _PK, _AAT, _ET],
    [0, 21, 0, 4, 0], 3
];
exports._Record$ = _Record$;
const RegisterStreamConsumerInput$ = [3, n0, _RSCI,
    0,
    [_SARN, _CN, _SI, _T],
    [0, 0, 0, 128 | 0], 2
];
exports.RegisterStreamConsumerInput$ = RegisterStreamConsumerInput$;
const RegisterStreamConsumerOutput$ = [3, n0, _RSCO,
    0,
    [_C],
    [() => Consumer$], 1
];
exports.RegisterStreamConsumerOutput$ = RegisterStreamConsumerOutput$;
const RemoveTagsFromStreamInput$ = [3, n0, _RTFSI,
    0,
    [_TK, _SN, _SARN, _SI],
    [64 | 0, 0, 0, 0], 1
];
exports.RemoveTagsFromStreamInput$ = RemoveTagsFromStreamInput$;
const SequenceNumberRange$ = [3, n0, _SNR,
    0,
    [_SSN, _ESN],
    [0, 0], 1
];
exports.SequenceNumberRange$ = SequenceNumberRange$;
const Shard$ = [3, n0, _Sh,
    0,
    [_SIh, _HKR, _SNR, _PSI, _APSI],
    [0, () => HashKeyRange$, () => SequenceNumberRange$, 0, 0], 3
];
exports.Shard$ = Shard$;
const ShardFilter$ = [3, n0, _SF,
    0,
    [_Ty, _SIh, _Ti],
    [0, 0, 4], 1
];
exports.ShardFilter$ = ShardFilter$;
const SplitShardInput$ = [3, n0, _SSI,
    0,
    [_STS, _NSHK, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.SplitShardInput$ = SplitShardInput$;
const StartingPosition$ = [3, n0, _SP,
    0,
    [_Ty, _SNe, _Ti],
    [0, 0, 4], 1
];
exports.StartingPosition$ = StartingPosition$;
const StartStreamEncryptionInput$ = [3, n0, _SSEI,
    0,
    [_ET, _KI, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.StartStreamEncryptionInput$ = StartStreamEncryptionInput$;
const StopStreamEncryptionInput$ = [3, n0, _SSEIt,
    0,
    [_ET, _KI, _SN, _SARN, _SI],
    [0, 0, 0, 0, 0], 2
];
exports.StopStreamEncryptionInput$ = StopStreamEncryptionInput$;
const StreamDescription$ = [3, n0, _SD,
    0,
    [_SN, _SARN, _SSt, _S, _HMSa, _RPH, _SCT, _EMn, _SMD, _ET, _KI],
    [0, 0, 0, () => ShardList, 2, 1, 4, () => EnhancedMonitoringList, () => StreamModeDetails$, 0, 0], 8
];
exports.StreamDescription$ = StreamDescription$;
const StreamDescriptionSummary$ = [3, n0, _SDS,
    0,
    [_SN, _SARN, _SSt, _RPH, _SCT, _EMn, _OSC, _SI, _SMD, _ET, _KI, _CC, _WT, _MRSIKB],
    [0, 0, 0, 1, 4, () => EnhancedMonitoringList, 1, 0, () => StreamModeDetails$, 0, 0, 1, () => WarmThroughputObject$, 1], 7
];
exports.StreamDescriptionSummary$ = StreamDescriptionSummary$;
const StreamModeDetails$ = [3, n0, _SMD,
    0,
    [_SM],
    [0], 1
];
exports.StreamModeDetails$ = StreamModeDetails$;
const StreamSummary$ = [3, n0, _SStr,
    0,
    [_SN, _SARN, _SSt, _SMD, _SCT],
    [0, 0, 0, () => StreamModeDetails$, 4], 3
];
exports.StreamSummary$ = StreamSummary$;
const SubscribeToShardEvent$ = [3, n0, _STSE,
    0,
    [_R, _CSN, _MBL, _CSh],
    [() => RecordList, 0, 1, () => ChildShardList], 3
];
exports.SubscribeToShardEvent$ = SubscribeToShardEvent$;
const SubscribeToShardInput$ = [3, n0, _STSI,
    0,
    [_CARN, _SIh, _SP, _SI],
    [0, 0, () => StartingPosition$, 0], 3
];
exports.SubscribeToShardInput$ = SubscribeToShardInput$;
const SubscribeToShardOutput$ = [3, n0, _STSO,
    0,
    [_ES],
    [[() => SubscribeToShardEventStream$, 0]], 1
];
exports.SubscribeToShardOutput$ = SubscribeToShardOutput$;
const Tag$ = [3, n0, _Ta,
    0,
    [_K, _V],
    [0, 0], 1
];
exports.Tag$ = Tag$;
const TagResourceInput$ = [3, n0, _TRI,
    0,
    [_T, _RARN, _SI],
    [128 | 0, 0, 0], 2
];
exports.TagResourceInput$ = TagResourceInput$;
const UntagResourceInput$ = [3, n0, _URI,
    0,
    [_TK, _RARN, _SI],
    [64 | 0, 0, 0], 2
];
exports.UntagResourceInput$ = UntagResourceInput$;
const UpdateAccountSettingsInput$ = [3, n0, _UASI,
    0,
    [_MTBC],
    [() => MinimumThroughputBillingCommitmentInput$], 1
];
exports.UpdateAccountSettingsInput$ = UpdateAccountSettingsInput$;
const UpdateAccountSettingsOutput$ = [3, n0, _UASO,
    0,
    [_MTBC],
    [() => MinimumThroughputBillingCommitmentOutput$]
];
exports.UpdateAccountSettingsOutput$ = UpdateAccountSettingsOutput$;
const UpdateMaxRecordSizeInput$ = [3, n0, _UMRSI,
    0,
    [_MRSIKB, _SARN, _SI],
    [1, 0, 0], 1
];
exports.UpdateMaxRecordSizeInput$ = UpdateMaxRecordSizeInput$;
const UpdateShardCountInput$ = [3, n0, _USCI,
    0,
    [_TSC, _ST, _SN, _SARN, _SI],
    [1, 0, 0, 0, 0], 2
];
exports.UpdateShardCountInput$ = UpdateShardCountInput$;
const UpdateShardCountOutput$ = [3, n0, _USCO,
    0,
    [_SN, _CSC, _TSC, _SARN],
    [0, 1, 1, 0]
];
exports.UpdateShardCountOutput$ = UpdateShardCountOutput$;
const UpdateStreamModeInput$ = [3, n0, _USMI,
    0,
    [_SARN, _SMD, _SI, _WTMB],
    [0, () => StreamModeDetails$, 0, 1], 2
];
exports.UpdateStreamModeInput$ = UpdateStreamModeInput$;
const UpdateStreamWarmThroughputInput$ = [3, n0, _USWTI,
    0,
    [_WTMB, _SARN, _SN, _SI],
    [1, 0, 0, 0], 1
];
exports.UpdateStreamWarmThroughputInput$ = UpdateStreamWarmThroughputInput$;
const UpdateStreamWarmThroughputOutput$ = [3, n0, _USWTO,
    0,
    [_SARN, _SN, _WT],
    [0, 0, () => WarmThroughputObject$]
];
exports.UpdateStreamWarmThroughputOutput$ = UpdateStreamWarmThroughputOutput$;
const WarmThroughputObject$ = [3, n0, _WTO,
    0,
    [_TMB, _CMB],
    [1, 1]
];
exports.WarmThroughputObject$ = WarmThroughputObject$;
var __Unit = "unit";
var ChildShardList = [1, n0, _CSL,
    0, () => ChildShard$
];
var ConsumerList = [1, n0, _CL,
    0, () => Consumer$
];
var EnhancedMonitoringList = [1, n0, _EML,
    0, () => EnhancedMetrics$
];
var MetricsNameList = 64 | 0;
var PutRecordsRequestEntryList = [1, n0, _PRREL,
    0, () => PutRecordsRequestEntry$
];
var PutRecordsResultEntryList = [1, n0, _PRRELu,
    0, () => PutRecordsResultEntry$
];
var RecordList = [1, n0, _RL,
    0, () => _Record$
];
var ShardIdList = 64 | 0;
var ShardList = [1, n0, _SLh,
    0, () => Shard$
];
var StreamNameList = 64 | 0;
var StreamSummaryList = [1, n0, _SSL,
    0, () => StreamSummary$
];
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var TagMap = 128 | 0;
const SubscribeToShardEventStream$ = [4, n0, _STSES,
    { [_st]: 1 },
    [_STSE, _RNFE, _RIUE, _KMSDE, _KMSISE, _KMSADE, _KMSNFE, _KMSOIR, _KMSTE, _IFE],
    [() => SubscribeToShardEvent$, [() => ResourceNotFoundException$, 0], [() => ResourceInUseException$, 0], [() => KMSDisabledException$, 0], [() => KMSInvalidStateException$, 0], [() => KMSAccessDeniedException$, 0], [() => KMSNotFoundException$, 0], [() => KMSOptInRequired$, 0], [() => KMSThrottlingException$, 0], [() => InternalFailureException$, 0]]
];
exports.SubscribeToShardEventStream$ = SubscribeToShardEventStream$;
exports.AddTagsToStream$ = [9, n0, _ATTS,
    0, () => AddTagsToStreamInput$, () => __Unit
];
exports.CreateStream$ = [9, n0, _CSr,
    0, () => CreateStreamInput$, () => __Unit
];
exports.DecreaseStreamRetentionPeriod$ = [9, n0, _DSRP,
    0, () => DecreaseStreamRetentionPeriodInput$, () => __Unit
];
exports.DeleteResourcePolicy$ = [9, n0, _DRP,
    0, () => DeleteResourcePolicyInput$, () => __Unit
];
exports.DeleteStream$ = [9, n0, _DS,
    0, () => DeleteStreamInput$, () => __Unit
];
exports.DeregisterStreamConsumer$ = [9, n0, _DSC,
    0, () => DeregisterStreamConsumerInput$, () => __Unit
];
exports.DescribeAccountSettings$ = [9, n0, _DAS,
    0, () => DescribeAccountSettingsInput$, () => DescribeAccountSettingsOutput$
];
exports.DescribeLimits$ = [9, n0, _DL,
    0, () => DescribeLimitsInput$, () => DescribeLimitsOutput$
];
exports.DescribeStream$ = [9, n0, _DSe,
    0, () => DescribeStreamInput$, () => DescribeStreamOutput$
];
exports.DescribeStreamConsumer$ = [9, n0, _DSCe,
    0, () => DescribeStreamConsumerInput$, () => DescribeStreamConsumerOutput$
];
exports.DescribeStreamSummary$ = [9, n0, _DSS,
    0, () => DescribeStreamSummaryInput$, () => DescribeStreamSummaryOutput$
];
exports.DisableEnhancedMonitoring$ = [9, n0, _DEM,
    0, () => DisableEnhancedMonitoringInput$, () => EnhancedMonitoringOutput$
];
exports.EnableEnhancedMonitoring$ = [9, n0, _EEM,
    0, () => EnableEnhancedMonitoringInput$, () => EnhancedMonitoringOutput$
];
exports.GetRecords$ = [9, n0, _GR,
    0, () => GetRecordsInput$, () => GetRecordsOutput$
];
exports.GetResourcePolicy$ = [9, n0, _GRP,
    0, () => GetResourcePolicyInput$, () => GetResourcePolicyOutput$
];
exports.GetShardIterator$ = [9, n0, _GSI,
    0, () => GetShardIteratorInput$, () => GetShardIteratorOutput$
];
exports.IncreaseStreamRetentionPeriod$ = [9, n0, _ISRP,
    0, () => IncreaseStreamRetentionPeriodInput$, () => __Unit
];
exports.ListShards$ = [9, n0, _LS,
    0, () => ListShardsInput$, () => ListShardsOutput$
];
exports.ListStreamConsumers$ = [9, n0, _LSC,
    0, () => ListStreamConsumersInput$, () => ListStreamConsumersOutput$
];
exports.ListStreams$ = [9, n0, _LSi,
    0, () => ListStreamsInput$, () => ListStreamsOutput$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceInput$, () => ListTagsForResourceOutput$
];
exports.ListTagsForStream$ = [9, n0, _LTFS,
    0, () => ListTagsForStreamInput$, () => ListTagsForStreamOutput$
];
exports.MergeShards$ = [9, n0, _MS,
    0, () => MergeShardsInput$, () => __Unit
];
exports.PutRecord$ = [9, n0, _PR,
    0, () => PutRecordInput$, () => PutRecordOutput$
];
exports.PutRecords$ = [9, n0, _PRu,
    0, () => PutRecordsInput$, () => PutRecordsOutput$
];
exports.PutResourcePolicy$ = [9, n0, _PRP,
    0, () => PutResourcePolicyInput$, () => __Unit
];
exports.RegisterStreamConsumer$ = [9, n0, _RSC,
    0, () => RegisterStreamConsumerInput$, () => RegisterStreamConsumerOutput$
];
exports.RemoveTagsFromStream$ = [9, n0, _RTFS,
    0, () => RemoveTagsFromStreamInput$, () => __Unit
];
exports.SplitShard$ = [9, n0, _SSp,
    0, () => SplitShardInput$, () => __Unit
];
exports.StartStreamEncryption$ = [9, n0, _SSE,
    0, () => StartStreamEncryptionInput$, () => __Unit
];
exports.StopStreamEncryption$ = [9, n0, _SSEt,
    0, () => StopStreamEncryptionInput$, () => __Unit
];
exports.SubscribeToShard$ = [9, n0, _STSu,
    0, () => SubscribeToShardInput$, () => SubscribeToShardOutput$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => TagResourceInput$, () => __Unit
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceInput$, () => __Unit
];
exports.UpdateAccountSettings$ = [9, n0, _UAS,
    0, () => UpdateAccountSettingsInput$, () => UpdateAccountSettingsOutput$
];
exports.UpdateMaxRecordSize$ = [9, n0, _UMRS,
    0, () => UpdateMaxRecordSizeInput$, () => __Unit
];
exports.UpdateShardCount$ = [9, n0, _USC,
    0, () => UpdateShardCountInput$, () => UpdateShardCountOutput$
];
exports.UpdateStreamMode$ = [9, n0, _USM,
    0, () => UpdateStreamModeInput$, () => __Unit
];
exports.UpdateStreamWarmThroughput$ = [9, n0, _USWT,
    0, () => UpdateStreamWarmThroughputInput$, () => UpdateStreamWarmThroughputOutput$
];
